analysis/dataset_clean/fn-ref.R

```diff
@@ -0,0 +1,87 @@
+# Function to set reference levels for factors
+ref <- function(input) {
+  
+  # Handle missing values --------------------------------------------------------
+  print('Handle missing values')
+  
+  input$cov_cat_smoking <- replace(input$cov_cat_smoking, 
+                                          is.na(input$cov_cat_smoking),
+                                          "M")
+  
+  # Set reference levels for factors ---------------------------------------------
+  print('Set reference levels for factors')
+  
+  cat_factors <- colnames(input)[grepl("_cat_",colnames(input))]
+  input[,cat_factors] <- lapply(input[,cat_factors], function(x) factor(x, ordered = FALSE))
+  
+  # Set reference level for variable: sub_cat_covidhospital -------------------
+  print('Set reference level for variable: sub_cat_covidhospital')
+  
+  input$sub_cat_covidhospital <- ordered(input$sub_cat_covidhospital, 
```

Comment:
Why this order? Escalating severity would be "no_infection" > "non_hospitalised" > "hospitalised".